src/py-opentimelineio/opentimelineio/adapters/filebundle.py

```diff
@@ -0,0 +1,123 @@
+"""OTIOZ adapter - bundles otio files linked to local media
+
+Takes as input an OTIO file that has media references which are all relative
+local paths (ie file:///foo.mov) and bundles those files and the otio file into
+a single zip file with the suffix .otioz.  Can error out if files aren't
+locally referenced or provide missing references
+"""
+
+import os
+import copy
+import zipfile
+
+from .. import (
+    exceptions,
+    schema,
+)
+
+from . import (
+    otio_json
+)
+
+
+class NotAFileOnDisk(exceptions.OTIOError):
+    pass
+
+
+def read_from_file(filepath):
+    if not zipfile.is_zipfile(filepath):
+        raise exceptions.OTIOError("Not a zipfile: {}".format(filepath))
+
+    internal_playlist_file = "{0}/{0}.otio".format(
+        os.path.splitext(os.path.basename(filepath))[0]
+    )
+    with zipfile.ZipFile(filepath, 'r') as zi:
+        result = otio_json.read_from_string(zi.read(internal_playlist_file))
+
+    return result
+
+
+class MediaReferencePolicy:
+    ErrorIfNotFile = 0
+    MissingIfNotFile = 1
+
+
+def write_to_file(
+    input_otio,
+    filepath,
+    unreachable_media_policy=MediaReferencePolicy.ErrorIfNotFile,
+    dryrun=False
+):
+    referenced_files = set()
+
+    for cl in input_otio.each_clip():
+        try:
+            target_file = cl.media_reference.target_url
+        except AttributeError:
+            continue
+
+        if target_file in referenced_files:
+            continue
+
+        if not os.path.exists(target_file) or not os.path.isfile(target_file):
+            if unreachable_media_policy is MediaReferencePolicy.ErrorIfNotFile:
+                raise NotAFileOnDisk(target_file)
+            if unreachable_media_policy is MediaReferencePolicy.MissingIfNotFile:
+                md = copy.deepcopy(cl.media_reference)
+                cl.media_reference = schema.MissingReference(
+                    name=md.name,
+                    metadata={
+                        'OTIOZ': {
+                            'original_reference': md
+                        }
+                    }
+                )
+                continue
+
+        referenced_files.add(target_file)
+
+    # dryrun reports the total size of files
+    if dryrun:
+        fsize = 0
+        for fn in referenced_files:
+            fsize += os.path.getsize(fn)
+        return fsize
+
+    fmapping = {}
+
+    # strip off the [z] suffix
+    otio_basename = os.path.splitext(os.path.basename(filepath))[0]
+
+    # gather the files up in the staging_dir
+    for fn in referenced_files:
+        target = os.path.join(otio_basename, os.path.basename(fn))
+        fmapping[fn] = target
+
+    # update the media references
+    for cl in input_otio.each_clip():
+        try:
+            source_fpath = cl.metadata.target_url
```

Comment:
This looks like it's setting a target_url property on the clip metadata. Shouldn't this be operating on the media_reference?